pkg/github/comment_service.go

```diff
@@ -1,82 +1,16 @@
 package github
 
 import (
-	"fmt"
-	"strings"
-
-	"github.com/arquillian/ike-prow-plugins/pkg/log"
 	"github.com/arquillian/ike-prow-plugins/pkg/scm"
-	"github.com/arquillian/ike-prow-plugins/pkg/utils"
-)
-
-// PluginTitleTemplate is a constant template containing "Ike Plugins (name-of-plugin)" title with markdown formatting
-const (
-	PluginTitleTemplate     = "### Ike Plugins (%s)"
-	assigneeMentionTemplate = "Thank you @%s for this contribution!"
 )
 
-// CommentService is a struct managing plugin comments
+// CommentService is a struct managing issue or pull request comments
 type CommentService struct {
-	client         *Client
-	log            log.Logger
-	issue          scm.RepositoryIssue
-	commentContext CommentContext
-}
-
-// CommentContext holds a plugin name and a assignee to be mentioned in the comment
-type CommentContext struct {
-	PluginName string
-	Assignee   string // TODO rethink this naming when plugins will start interacting with issue creators and reviewers
-}
-
-// NewCommentService creates an instance of GitHub CommentService for the given CommentContext
-func NewCommentService(client *Client, log log.Logger, change scm.RepositoryChange, issueOrPrNumber int, commentContext CommentContext) *CommentService {
-	return &CommentService{
-		client: client,
-		log:    log,
-		issue: scm.RepositoryIssue{
-			Owner:    change.Owner,
-			RepoName: change.RepoName,
-			Number:   issueOrPrNumber,
-		},
-		commentContext: commentContext,
-	}
-}
-
-// PluginComment checks all present comments in the issue/pull-request. If no comment with PluginTitleTemplate
-// (with the related plugin) is found, then it adds a new comment with the plugin title, assignee mention
-// and the given commentMsg. If such a comment is present already, then it does nothing.
-func (s *CommentService) PluginComment(commentMsg string) error {
-
-	comments, err := s.client.ListIssueComments(s.issue)
-	if err != nil {
-		s.log.Errorf("Getting all comments failed with an error: %s", err)
-	}
-
-	for _, com := range comments {
-		content := *com.Body
-		if strings.HasPrefix(content, s.getPluginTitle()) {
-			return nil
-		}
-	}
-
-	err = s.client.CreateIssueComment(s.issue, s.createPluginHint(commentMsg))
-
-	return err
-}
-
-func (s *CommentService) append(first, second string) string {
-	return first + "\n\n" + second
-}
-
-func (s *CommentService) createPluginHint(commentMsg string) *string {
-	return utils.String(s.append(s.createBeginning(), commentMsg))
-}
-
-func (s *CommentService) createBeginning() string {
-	return s.append(s.getPluginTitle(), fmt.Sprintf(assigneeMentionTemplate, s.commentContext.Assignee))
+	client *Client
+	issue  scm.RepositoryIssue
 }
 
-func (s *CommentService) getPluginTitle() string {
-	return fmt.Sprintf(PluginTitleTemplate, s.commentContext.PluginName)
+// Comment adds a comment message to the issue
+func (s *CommentService) Comment(commentMsg *string) error {
+	return s.client.CreateIssueComment(s.issue, commentMsg)
```

Comment:
If the only thing this service does now is to delegate to `CreateIssueComment` from our `go-github` facade, do we really need it?

Reply:
> plugin should comment that a user is not allowed to use some command

Not sure if we shouldn't just ignore such case. Will easily become a problem of running out of API call limits.

Reply:
well, I think that the number of the not-allowed attempts won't be too much. And, from my point of view, it is much better to notify/comment why the command is not working for such a user than just to stay silent. The current comment msg that is sent looks like:

> @bartoszmajsak-test has used a command `/ok-without-tests` but this user is not allowed to do that. User with the necessary permissions is anybody who is admin or requested reviewer, but not pull request creator

This comment is generated automatically based on the information that are set for the specific command...

Reply:
We have 5k API calls per hour, right? Multiply it per repos, prs and edits on them. I would be very careful with extra comments, due to this fact. It also can be gamed by anyone having account on GH and we can easily put the service down in no time.

If we list all possible commands and roles allowed in the hint in `<details>` section, for example, we have it all explained without a need of repeating it over and over again.

Reply:
Moved the discussion here https://github.com/arquillian/ike-prow-plugins/issues/73#issuecomment-378845901